### DISCUSSION TITLE
Why trivy takes long time to scan the cluster?

### DISCUSSION THREAD

**Discussion by nilesh-akhade** [aquasecurity/trivy#2829] [Q&A] [answered]
The trivy cluster scan command(`trivy k8s cluster`) takes 30-45 mins to complete. The starboard CLI, however, takes 10-15 mins on the same cluster. Is it because trivy runs so many checks?

This question is **not about comparison** with other tools. But would like to know what goes behind the scene, when trivy runs a cluster scan?

**Comment by nilesh-akhade** [comment, marked as answer]
I am closing this with my own answer.
- Trivy runs so many checks for misconfiguration, like 6 lakh misconfiguration checks for a sufficiently small cluster
- Trivy image scan is pivoted around Kubernetes resource which duplicates the vulnerability results for every k8s resource if it is using the same image.

**Comment by cap0** [comment]
any way to speed up it? Maybe removing misconfiguration checks.

**Reply by DmitriyLewen** [reply]
Hello @cap0 
You can use `--scanners` flag to enable only required scanners.
example for vulnerabilities- https://aquasecurity.github.io/trivy/v0.47/docs/target/kubernetes/#control-plane-and-node-components-vulnerability-scanning

**Reply by cap0** [reply]
Thank you it is super fast now!
